homeassistant/components/islamic_prayer_times/__init__.py

```diff
@@ -1 +1,206 @@
 """The islamic_prayer_times component."""
+from datetime import datetime, timedelta
+import logging
+
+from prayer_times_calculator import PrayerTimesCalculator, exceptions
+import voluptuous as vol
+
+from homeassistant.config_entries import SOURCE_IMPORT
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.event import async_track_point_in_time
+import homeassistant.util.dt as dt_util
+
+from .const import (
+    CALC_METHODS,
+    CONF_CALC_METHOD,
+    DATA_UPDATED,
+    DEFAULT_CALC_METHOD,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: {
+            vol.Optional(CONF_CALC_METHOD, default=DEFAULT_CALC_METHOD): vol.In(
+                CALC_METHODS
+            ),
+        }
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass, config):
+    """Import the Islamic Prayer component from config."""
+    if DOMAIN in config:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": SOURCE_IMPORT}, data=config[DOMAIN]
+            )
+        )
+
+    return True
+
+
+async def async_setup_entry(hass, config_entry):
+    """Set up the Islamic Prayer Component."""
+    client = IslamicPrayerClient(hass, config_entry)
+
+    if not await client.async_setup():
+        return False
+
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN] = client
+    return True
+
+
+async def async_unload_entry(hass, config_entry):
+    """Unload Transmission Entry from config_entry."""
+
+    await hass.config_entries.async_forward_entry_unload(config_entry, "sensor")
+
+    hass.data.pop(DOMAIN)
+
+    return True
+
+
+class IslamicPrayerClient:
```

Comment:
Consider breaking this out into a new file.

Reply:
Usually I keep such class under __init__.py if it is a simple client class. If it is recommended to move it to new a file I can do that.